### PR TITLE
Avoid watching ignored folders on callbacks (#805)

### LINKED ISSUE #805: lsp-file-watch-ignored  is not sufficient to prevent file notifications
Given the following project structure:

```
root/
  src/
    app/
      .project
      src/ <- should be watched
      tst/ <- ditto
      build/ <- should be ignored
```

Adding `build` to the ignored files does not avoid the file notifications.

(cc @aureumleonis )

### PR DESCRIPTION
**Pull request by aureumleonis**
Fixes #805 by also applying the filer on the folder created event

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -1071,24 +1071,7 @@ already have been created."
...
-                  ;; process the files that are already present in
-                  ;; the directory.
-                  (->> (directory-files-recursively file-name ".*" t)
-                       (seq-do (lambda (f)
-                                 (unless (file-directory-p f)
-                                   (funcall callback (list nil 'created f)))))))
-                 ((and (not (file-directory-p file-name))
-                       (memq event-type '(created deleted changed)))
-                  (funcall callback event))))))
+            (lambda (event) (funcall 'lsp--folder-watch-callback event callback watch)))
```
Comment (yyoncho) [inline lsp-mode.el]: You can directly call lsp--folder-watch-callback without funcall

**Comment by yyoncho** [comment]
Any update?

**Comment by aureumleonis** [comment]
I've been on holiday... I can get back to this maybe next week

**Comment by yyoncho** [comment]
Merging as it is, the comments can be addressed separately.

**Pull request merged**